### DISCUSSION TITLE
Optional file upload not working?

### DISCUSSION THREAD

**Discussion by yannickl88** [swaggest/rest#159] [Q&A] [answered]
Hey, cool package, really love that if feels like a complete solution.

I was messing around with some of the input definitions. However, I think it is not possible to have a optional file uploads.

```golang
func Upload() usecase.Interactor {
	type Input struct {
		Upload *multipart.FileHeader `formData:"file"`
		Url    string                `formData:"url"`
	}

	u := usecase.NewInteractor(func(ctx context.Context, input Input, output *string) error {
		// ...

		return nil
	})
	u.SetDescription("Upload a file or URL")
	u.SetTags("Default")

	return u
}
```

None of the input fields are required, however, if I post something with only the `url` I get a 400 Bad Request and the following result:
```json
{
  "status": "INVALID_ARGUMENT",
  "error": "invalid argument: failed to get file \"file\" from request: http: no such file"
}
```

Am I missing something or did I hit a bug?

**Comment by vearutop** [comment, marked as answer]
Thank you for reporting this, indeed it is a bug (fixed in https://github.com/swaggest/rest/pull/160).

**Reply by yannickl88** [reply]
Thanks for the quick fix!
